[LLM] Gemma finetuning example with LoRA (#3238)
skypilot-org/skypilot: llm/gemma/README.md

      "max_tokens": 25
  }'
```

## Finetune Gemma on Your Own Data

> You can also finetune Gemma on your own data with SkyPilot as well. We have an example LoRA finetuning script in [scripts/finetune-lora.py](scripts/finetune-lora.py), where we finetune the model on `Abirate/english_quotes` dataset.

Reviewer: ```suggestion
You can also finetune Gemma on your own data with SkyPilot. We provide an example LoRA finetuning script in [scripts/finetune-lora.py](scripts/finetune-lora.py), where the model is finetuned on  the `Abirate/english_quotes` dataset.
```

```

You will get the following output:
```
{'loss': 1.3736, 'grad_norm': 3.4177560806274414, 'learning_rate': 0.0001, 'epoch': 1.0}
{'loss': 0.4876, 'grad_norm': 1.5821903944015503, 'learning_rate': 0.0002, 'epoch': 1.33}
{'loss': 0.6929, 'grad_norm': 2.109269380569458, 'learning_rate': 0.000175, 'epoch': 2.0}
{'loss': 0.4633, 'grad_norm': 1.5081373453140259, 'learning_rate': 0.00015000000000000001, 'epoch': 2.67}
{'loss': 0.2498, 'grad_norm': 0.8380517959594727, 'learning_rate': 0.000125, 'epoch': 3.0}
{'loss': 0.5513, 'grad_norm': 1.2455085515975952, 'learning_rate': 0.0001, 'epoch': 4.0}
{'loss': 0.4849, 'grad_norm': 1.32512366771698, 'learning_rate': 7.500000000000001e-05, 'epoch': 5.0}
{'loss': 0.1754, 'grad_norm': 1.753520131111145, 'learning_rate': 5e-05, 'epoch': 5.33}
{'loss': 0.2929, 'grad_norm': 1.5430747270584106, 'learning_rate': 2.5e-05, 'epoch': 6.0}
{'loss': 0.2797, 'grad_norm': 1.159482717514038, 'learning_rate': 0.0, 'epoch': 6.67}
{'train_runtime': 7.3703, 'train_samples_per_second': 5.427, 'train_steps_per_second': 1.357, 'train_loss': 0.505131047964096, 'epoch': 6.67}
100%|██████████| 10/10 [00:07<00:00,  1.35it/s]
```

> You can also save the cost (~3x) by using the same YAML to finetune the model on spot instances:
> ```bash
> WANDB_API_KEY="xxx" HF_TOKEN="xxx" sky spot launch -n gemma finetune-lora.yaml --env HF_TOKEN --env WANDB_API_KEY --env BUCKET_NAME=your-bucket-name
> ```

Reviewer: ```suggestion
You can also reduce cost by ~3x by using the same YAML to finetune the model on spot instances (auto-recovery enabled):
```bash
WANDB_API_KEY="xxx" HF_TOKEN="xxx" sky spot launch -n gemma finetune-lora.yaml --env HF_TOKEN --env WANDB_API_KEY --env BUCKET_NAME=your-bucket-name
```
```
